app/domain/authentication/authn_jwt/authenticator.rb

```diff
@@ -0,0 +1,143 @@
+require 'command_class'
+
+module Authentication
+  module AuthnJwt
+    # Generic JWT authenticator that receive JWT vendor configuration and uses to validate that the authentication
+    # request is valid, and return conjur authn token accordingly
+    Authenticator = CommandClass.new(
+      dependencies: {
+        token_factory: TokenFactory.new,
+        logger: Rails.logger,
+        audit_log: ::Audit.logger,
+        validate_origin: ::Authentication::ValidateOrigin.new,
+        role_class: ::Role,
+        webservice_class: ::Authentication::Webservice,
+        validate_role_can_access_webservice: ::Authentication::Security::ValidateRoleCanAccessWebservice.new,
+        role_id_class: Audit::Event::Authn::RoleId
+      },
+      inputs: %i[jwt_configuration authenticator_input]
```

Comment:
i'd like to challenge the name `jwt_configuration`. it is not only the name of course, i think that a better name will assist us in building the code in a cleaner way. it is clearly an object that will change according to the url/token and in a way it is some sort of `jwt_specfic_authenticator` which will perform some of the authentication stages differently (e.g `validate_and_decode_token`).

can we please challenge this and think of a better name? try to summarize its purpose in a sentence or a paragraph.

Reply:
vendor_configuration can be better name ? wanna talk about it?